ssl/statem/statem_clnt.c

```diff
@@ -1136,6 +1136,21 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
                session_id_len);
     }
 
+    if (!ssl_verify_client_session_version(s)) {
```

Comment:
So I'm confused about what it's trying to do.  I assume that this is the case of session resumption, and so `s->hit == 1`?  Are you trying to detect session resumption in some other way, and then check that the version is the one you expected?